### PR TITLE
Fix multiple minor styling issues

### LINKED ISSUE #23: Tab focus styling
Based on https://github.com/wagtail/wagtail/issues/6793

We need to discuss and define what we want.

### LINKED ISSUE #59: Implement sidebar according to design
https://www.figma.com/file/hFjWqQeEjGd8NtTsEHcc4i/Wagtail-Documentation-wagtail.io?node-id=1%3A2

Fold buttons
Underlines
Right border

### LINKED ISSUE #99: Max width for sidebar (main navigation)
Currently, the width of the sidebar (main navigation) increases indefinitely. 

This should be limited to some sensible value to make the layout work nicer on larger screens.

### LINKED ISSUE #107: Increase space after admonition code blocks
https://docs.wagtail.io/en/latest/topics/pages.html#inline-models

I think there should be more space between the code block and the text below it.

image

### LINKED ISSUE #190: sidebar menu - creating horizontal scroll bar
The first thing that jumps out is that the sidebar menu has a horizontal scroller.

Preview: https://wagtail--9013.org.readthedocs.build/en/9013/

_Originally posted by @zerolab in https://github.com/wagtail/wagtail/issues/9013#issuecomment-1217592734_

### PR DESCRIPTION
**Pull request by lb-**
Resolve a range of small styling issues across the documentation

- fix gap after content within admonitions, avoid styling every last element inside, only the container - fixes #107
- use variable for the max-width of the version picker (300px on bootstrap variables)
- add layout scss to fix some global style issues (already on Wagtail core)
- add .sidebar-container so that we can have a max width on navbar - fixes #99
- use .sidebar-container to allow for border-right on larger devices to match designs - fixes #59 (note: the [designs](https://www.figma.com/file/hFjWqQeEjGd8NtTsEHcc4i/Wagtail-Documentation-(wagtail.io)) have some additional tweaks but this gets is to 90% and we can consider the design adoption closed)
- resolve styling issues with sidebar container & overflow scroll - fixes #190
- avoid using coded styles when bootstrap utility classes are suitable in the sidebar & search box
- add high contrast focus outline to focusable (tabbable) elements - fixes #23 (note: colour is based on latest recommendation in wagtail core https://github.com/wagtail/wagtail/issues/8823#issuecomment-1212649398 )

### CODE REVIEW

**Comment by lb-** [comment]
Once this is adopted in Wagtail core in the future - we can remove some of the styling overrides in Wagtail docs.

https://github.com/wagtail/wagtail/blob/cc4675aa5ed41ffde68960fa697b3e3e682d71fc/docs/_static/css/custom.css#L1-L10

File: sphinx_wagtail_theme/theme.conf [hunk 1]
```diff
@@ -12,5 +12,5 @@ logo_height = 59
 logo_url = /
 logo_width = 45
 header_links =
-footer_links = Features|https://wagtail.io/features/, About Wagtail|https://wagtail.io/about-wagtail/, Services|https://wagtail.io/services/, Blog|https://wagtail.io/blog/, Packages|https://wagtail.io/packages/, Developers|https://wagtail.io/developers/
+footer_links = Features|https://wagtail.org/features/, About Wagtail|https://wagtail.org/about-wagtail/, Services|https://wagtail.org/services/, Blog|https://wagtail.org/blog/, Packages|https://wagtail.org/packages/, Developers|https://wagtail.org/developers/
```
Comment (vsalvino) [inline sphinx_wagtail_theme/theme.conf:15]: I would think the default here should be empty, and move these footer_links to Wagtail core's `docs/conf.py`

**Comment by lb-** [comment]
looks like I've accidentally included the links fixes in this PR also - will fix it up.

**Comment by vsalvino** [comment]
These changes look good to me. I haven't pulled the branch and rendered it, but noticed some of the same issues while poking around for other features.

**Comment by lb-** [comment]
Thanks - re-validated again and will merge this in.

**Pull request merged**